### PR TITLE
Add SWFFT backend from their Git repo

### PR DESCRIPTION
**Pull request by luszczek**
SWFFT is distributed in a Git repo. I added a settable variable `FIBER_SWFFT_REPO` that would allow to point to that repo so the files from SWFFT would be included during build. Some of their files are C++ so I had to add another language.
Also, SWFFT requires CPU version FFTW3 so both backends have to be enabled at the same time.
A sample build line looked like this for me:
```
 cmake -D FIBER_ENABLE_SWFFT=ON -DFIBER_ENABLE_FFTW=ON -D FIBER_FFT_LIB_DIRS="${FFTW_ROOT}/lib" -D FIBER_FFT_INCLUDE_DIRS="${FFTW_ROOT}/include;${SWFFT_REPO_ROOT}" -D MPI_DIR=${MPI_ROOT} -D FIBER_SWFFT_REPO=${SWFFT_REPO_ROOT} /path/to/fiber
```
All the environment variables with `_ROOT` suffix should be defined by the user to point to the installation locations.

### CODE REVIEW

**Comment by G-Ragghianti** [comment]
The CI says that there is a problem in the CMakefiles.txt.

File: benchmarks/CMakeLists.txt [hunk 1]
```diff
@@ -27,7 +27,7 @@ macro(fiber_add_benchmark fiber_benchmark)
         target_link_libraries(${fiber_benchmark} ffte)
     endif()
     if (FIBER_ENABLE_SWFFT)
-        target_link_libraries(${fiber_benchmark} swfft)
+        target_link_libraries(${fiber_benchmark} fiber_swfft)
```
Comment (af-ayala) [inline benchmarks/CMakeLists.txt:30]: We should keep original library name: SWFFT

File: benchmarks/fiber_swfft.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
+
+#include "fiber_swfft.h"
```
Comment (af-ayala) [inline benchmarks/fiber_swfft.cpp:2]: We should keep original library name: SWFFT and swfft.h

File: include/fiber_backend_swfft.h [hunk 3]
```diff
@@ -9,7 +9,9 @@
 #include <stdio.h>
 
 #if defined(FIBER_ENABLE_SWFFT)
-#include "swfft.h"
+#include "fiber_swfft.h"
```
Comment (af-ayala) [inline include/fiber_backend_swfft.h:12]: We should keep original library name: SWFFT

File: benchmarks/fiber_swfft.cpp (hunk 2 above)
Comment (luszczek) [reply benchmarks/fiber_swfft.cpp:2]: There is no "original library".  See above.

File: include/fiber_backend_swfft.h (hunk 3 above)
Comment (luszczek) [reply include/fiber_backend_swfft.h:12]: There is no "original library".  See above.

File: benchmarks/CMakeLists.txt (hunk 1 above)
Comment (luszczek) [reply benchmarks/CMakeLists.txt:30]: There is no "original library".  See above.

File: benchmarks/fiber_swfft.cpp (hunk 2 above)
Comment (af-ayala) [reply benchmarks/fiber_swfft.cpp:2]: What I mean is that we can keep the name SWFFT when building the library using https://git.cels.anl.gov/hacc/SWFFT, 
fiber_swfft would sound like a different implementation. Also, the name fiber is a placeholder for now.
Using swfft is also more compact and inline with internal enumeration access.

**Comment by luszczek** [comment]
Note that SWFFT is a bare Git repo with a handful of files. The included make script builds two specific binaries. SWFFT is not a standalone project or a library. Any integration we add is specific to our project and hence all naming has `fiber` prefix.

File: benchmarks/fiber_swfft.cpp (hunk 2 above)
Comment (luszczek) [reply benchmarks/fiber_swfft.cpp:2]: SWFFT is the original library. FIBER_SWFFT is the integration piece missing in the original code. We only own FIBER name space. All the SWFFT names live in HACC name space which is the name of their application. There is no SWFFT name anywhere in the code. Only their `README` file contains the name. There is no reason to pretend there is a header file called `swfft.h`. That's why we create our own `fiber_swfft.h` that we own and can modify at any time.

**Review by af-ayala — APPROVED** [review]

File: benchmarks/fiber_swfft.cpp (hunk 2 above)
Comment (af-ayala) [reply benchmarks/fiber_swfft.cpp:2]: OK, it makes sense.

**Pull request merged**